src/routes/versions.rs

```diff
@@ -63,17 +66,47 @@ pub async fn versions_get(
         .await
         .map_err(|e| ApiError::DatabaseError(e.into()))?;
 
-    let versions: Vec<models::mods::Version> = versions_data
-        .into_iter()
-        .filter_map(|v| v)
-        .map(convert_version)
-        .collect();
+    let mut versions = Vec::new();
+
+    for version_data in versions_data {
+        if let Some(version) = version_data {
+            let mut authorized = version.accepted;
+
+            if !authorized {
+                let user = get_user_from_headers(req.headers(), &**pool).await?;
```

Comment:
This should be moved to the top of the function so that it isn't queried every time, and should have error handling so that non logged in users won't just get an error if the list contains any unapproved mods.